### ISSUE TITLE
Could not attach to process with CustompipeName 'AzureFunctionsPSWorker'

### ISSUE THREAD

**Issue by 0xUnicorn** [microsoft/vscode-azurefunctions#2625]
I have a Windows10 20H2 workstation suddenly giving me this error message in Visual Studio Code and without launching my function properly. 
It did work for 2 weeks or so and I managed to create some PoC functions.Then it stopped working.

I then tried installing azure function core tools, PS7, SDKs 3.1/5.1 and VisualStudioCode (Azure extension) on my laptop with Win10 20H2 and that worked and it's still working this day.

My conclusion was it must be my win10 workstation which has an issue. 

So I created a new VM with a Win10 20H2 and installed PS7, SDK's, AzureFuncCoreTools, VisualStudioCode (Azure extension) and then tried creating a new project, and I get the exact same error when executing.

So now I have 2 Virtual Machines running Win10 20H2 and neither of them work when trying to run local powershell functions
I have 1 Lenovo laptop with Win10 20H2 and it works fine.

For troubleshooting I have tried this on my newly created Win10 test VM:
uninstalled Powershell 7.1 and installed Powershell 7.0
uninstalled .NETcore SDK 5.1

Trying to run my function directly from PowerShell without using VSCode returns this:
PS Z:\Folder1\ProjectRoot> func host start:

Azure Functions Core Tools
Core Tools Version:       3.0.3233 Commit hash: d1772f733802122a326fa696dd4c086292ec0171
Function Runtime Version: 3.0.15193.0

Let me note, before starting the configuration of the new Win10 VM, I tried a last time executing my function on the Win10 workstation (original) and after 10-15 minutes it actually started my HTTPTrigger function and it worked. When I tried executing it once again it returned the error message. 

All machines are running the exact same project, so it's the same .json files I'm referring to in the end.

Please let me know if you need more information

Thanks for your time. 

----------------------------------------------------------------------------------------------------------------------------

Machine info:

Win10 workstation (original):

Visual Studio Code: 1.52.1

PS> dotnet --info:
`.NET SDK (reflecting any global.json):
 Version:   5.0.102
 Commit:    71365b4d42

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.102\

Host (useful for support):
  Version: 5.0.2
  Commit:  cb5f173b96

.NET SDKs installed:
  3.1.405 [C:\Program Files\dotnet\sdk]
  5.0.102 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]`

PS> func --version:
3.0.3233

PS> $PSVersionTable:
Name                           Value
PSVersion                      7.1.1
PSEdition                      Core
GitCommitId                    7.1.1
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}       
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

---------------------------------------------------------------------------------------------------------------------------------------------------

Win10 workstation (new test machine):

Visual Studio Code: 1.52.1

PS> dotnet --info:
.NET Core SDK (reflecting any global.json):
 Version:   3.1.405     
 Commit:    65f9d75b1c  

Runtime Environment:    
 OS Name:     Windows   
 OS Version:  10.0.19042
 OS Platform: Windows   
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.405\

Host (useful for support):
  Version: 3.1.11
  Commit:  f5eceb8105

.NET Core SDKs installed:
  3.1.405 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

PS> func --version:
3.0.3233

PS> $PSVersionTable:
Name                           Value
PSVersion                      7.0.3
PSEdition                      Core
GitCommitId                    7.0.3
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}       
PSRemotingProtocolVersion      2.3    
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

---------------------------------------------------------------------------------------------------------------------------------------------------

Win10 Laptop:

Visual Studio Code: 1.52.1

PS> dotnet --info:
.NET SDK (reflecting any global.json):
 Version:   5.0.102
 Commit:    71365b4d42

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.102\

Host (useful for support):
  Version: 5.0.2
  Commit:  cb5f173b96

.NET SDKs installed:
  3.1.401 [C:\Program Files\dotnet\sdk]
  3.1.405 [C:\Program Files\dotnet\sdk]
  5.0.102 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

PS> func --version:
3.0.3233

PS> $PSVersionTable:
Name                           Value
PSVersion                      7.1.1
PSEdition                      Core
GitCommitId                    7.1.1
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

---------------------------------------------------------------------------------------------------------------------------------------------------

.vscode/launch.json:

{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Attach to PowerShell Functions",
      "type": "PowerShell",
      "request": "attach",
      "customPipeName": "AzureFunctionsPSWorker",
      "runspaceId": 1,
      "preLaunchTask": "func: host start"
    }
  ]
}

---------------------------------------------------------------------------------------------------------------------------------------------------

.vscode/tasks.json:

{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "func",
      "command": "host start",
      "problemMatcher": "$func-powershell-watch",
      "isBackground": true
    }
  ]
}

---------------------------------------------------------------------------------------------------------------------------------------------------

.vscode/settings.json:

{
  "azureFunctions.deploySubpath": ".",
  "azureFunctions.projectLanguage": "PowerShell",
  "azureFunctions.projectRuntime": "~3",
  "debug.internalConsoleOptions": "neverOpen"
}

---------------------------------------------------------------------------------------------------------------------------------------------------

/host.json:

{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      }
    }
  },
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[1.*, 2.0.0)"
  },
  "managedDependency": {
    "enabled": true
  }
}

**Comment by ejizba** [comment]
Sorry for the late response. Are you still having issues? If so, please debug your project and copy the logs from the "func host start" task here. I believe this error usually happens when there's an error in those logs. Also - how did you install the func cli (choco, npm, msi, etc.)?

**Comment by 0xUnicorn** [comment]
Hi, yes I still have the same issue.

How would you like me to debug? 
I get nowhere even when running `func host start --debug`
The only thing is this:

`Core Tools Version:       3.0.3233 Commit hash: d1772f733802122a326fa696dd4c086292ec0171`
`Function Runtime Version: 3.0.15193.0`

**Comment by ejizba** [comment]
You should start debugging in VS Code, not directly with the func cli. You can press F5 or click the play button next to "Attach to PowerShell Functions"
<img width="451" alt="Screen Shot 2021-02-08 at 9 38 25 AM" src="https://user-images.githubusercontent.com/11282622/107259544-f7c74a80-69f1-11eb-8df2-d495f79f4960.png">

It will start the func cli for you and you should see logs in the "Terminal" pane, which are the logs I'm talking about:
<img width="877" alt="Screen Shot 2021-02-08 at 9 41 33 AM" src="https://user-images.githubusercontent.com/11282622/107259580-057cd000-69f2-11eb-9fa3-f493d000731c.png">

**Comment by 0xUnicorn** [comment]
Same result as running in the Terminal, except for a dialog box when I hit ctrl+c after n minutes.

Let me just remind you, this also happens on a cleaned installed VM with same configuration, and it did work well back in december/january. 

![Screenshot from 2021-02-09 10-11-42](https://user-images.githubusercontent.com/17219001/107341507-60302d80-6abf-11eb-987f-cbf862632e1d.png)

![Screenshot from 2021-02-09 10-13-00](https://user-images.githubusercontent.com/17219001/107341564-6f16e000-6abf-11eb-98c4-c5440cf4cd42.png)

Looking forward to your next suggestion.

**Comment by ejizba** [comment]
Hmm well that's not very helpful since it doesn't show much of anything in the logs. And that's exactly what it looks like if you run `func host start` manually from the command line? You could try running `func host start --verbose` to see if you get anything more helpful.

Also do you have this problem with a brand new, clean project with just an http trigger?

**Comment by 0xUnicorn** [comment]
This is what it looks like if I'm running `func host start` from the PWSH terminal. 

![image](https://user-images.githubusercontent.com/17219001/107512062-9a282f00-6ba6-11eb-9606-a4e3fe25d986.png)

This is quite interesting. When I created a new folder, initialized a new project with the VSC Azure extension inside, and added a HTTPTrigger function, it would work the first time when running in Debug (VSC - F5). Stopping it and running it again will give me the same output as posted in my last post.

So, I thougt creating a new project would work the first time again, so I could send you a screendump. But nope, another brand new folder with a clean project and only a HTTPTrigger running in Debug will give me the exact same output again. Even when restarting the OS in between.

Running the project with `func host start --verbose` gives me this:
![image](https://user-images.githubusercontent.com/17219001/107513372-68b06300-6ba8-11eb-8a31-58a29d92093c.png)

**Comment by 0xUnicorn** [comment]
Do you know if it's possible for me to raise a ticket within Azure or somewhere else for getting hands-on support on this issue, I've been postponing all of my powershell function app development as long as I could in the hopes of getting a fix for this issue.

Now I'm at a point where my backlog mostly contain powershell function app development and I have no other choice than starting development on my laptop which is by no means an ideal situation.

FYI. I'm a registered Microsoft Partner and doing a SaaS product for azure hosted on azure, if that changes anything in the possibility for getting hands-on support with your open-source products.

Thanks in advance.

**Comment by ejizba** [comment]
Well since the func cli shows the same thing in VS Code and the terminal (not much at all haha), it's probably a good idea to file an issue on the cli directly: https://github.com/Azure/azure-functions-core-tools/issues. I think the root cause is the func cli can't start up your app for some reason, and that's why VS Code can't attach.

I'm honestly not familiar with how being a "registered Microsoft Partner" works. You're more than welcome to file a ticket wherever applicable. The company probably has dedicated people to help on top of whatever discussion you get from my team on GitHub.

**Comment by 0xUnicorn** [comment]
Oh that's my bad! I haven't even noticed this was the vscode-azurefunctions and not the core tools. For some reason I got those mixed up in my head.
I'll try to open a new issue over there refrencing to this case. So feel free to close this one. 

Okay thanks for your input. I'll try open a support case in the Azure portal too.

Thank you for your time!
